src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Unit.Tests/GeneratedComInterfaceAnalyzerTests.cs

```diff
@@ -0,0 +1,758 @@
+            }
+
+            [Fact]
+            public async Task IDispatchShort()
+            {
+                string snippet =
+                    $$$"""
+
+                [InterfaceTypeAttribute((short)2)]
+                interface IFoo
+                {
+                    void Bar() {}
+                }
+
+                public partial class MyComWrappers : GeneratedComWrappersBase<ComObject>
+                {
+                }
+
+                """;
+                await VerifyCS.VerifyAnalyzerAsync(_usings + snippet);
+            }
+
+            [Fact]
+            public async Task IInspectable()
+            {
+                string snippet =
+                    $$$"""
+
+                [InterfaceTypeAttribute(ComInterfaceType.InterfaceIsIInspectable)]
+                interface IFoo
+                {
+                    void Bar() {}
+                }
+
+                public partial class MyComWrappers : GeneratedComWrappersBase<ComObject>
+                {
+                }
+
+                """;
+                await VerifyCS.VerifyAnalyzerAsync(_usings + snippet);
+            }
+
+            [Fact]
+            public async Task IInspectableShort()
+            {
+                string snippet =
+                    $$$"""
+
+                [InterfaceTypeAttribute((short)3)]
+                interface IFoo
+                {
+                    void Bar() {}
+                }
+
+                public partial class MyComWrappers : GeneratedComWrappersBase<ComObject>
+                {
+                }
+
+                """;
+                await VerifyCS.VerifyAnalyzerAsync(_usings + snippet);
+            }
+
+            [Fact]
+            public async Task IDual()
+            {
+                string snippet =
+                    $$$"""
+
+                [InterfaceTypeAttribute(ComInterfaceType.InterfaceIsDual)]
+                interface IFoo
+                {
+                    void Bar() {}
+                }
+
+                public partial class MyComWrappers : GeneratedComWrappersBase<ComObject>
+                {
+                }
+
+                """;
+                await VerifyCS.VerifyAnalyzerAsync(_usings + snippet);
+            }
+
+            [Fact]
+            public async Task IDualShort()
+            {
+                string snippet =
+                    $$$"""
+
+                [InterfaceTypeAttribute((short)0)]
+                interface IFoo
+                {
+                    void Bar() {}
+                }
+
+                public partial class MyComWrappers : GeneratedComWrappersBase<ComObject>
+                {
+                }
+
+                """;
+                await VerifyCS.VerifyAnalyzerAsync(_usings + snippet);
+            }
+        }
+
+    [ActiveIssue("https://github.com/dotnet/runtime/issues/60650", TestRuntimes.Mono)]
+        public class InterfaceHasGeneratedComInterfaceAttributeOnly
```

Comment:
All of the tests in this class are the same since the `InterfaceTypeAttribute` is not included. Can we collapse these down to one test for `[GeneratedComInterface]` without any `InterfaceTypeAttribute`?